packages/web/src/store/configureStore.ts

```diff
@@ -116,28 +116,41 @@ const middlewares = applyMiddleware(
 )
 
 // As long as the mobile client is dependent on the web client, we need to sync
-// the client store from web -> mobile
-const clientStoreKeys = Object.keys(commonStoreReducers)
+// the common store from web -> mobile
+const commonStateKeys = Object.keys(commonStoreReducers)
 
-const syncClientStateToNativeMobile = (store: Store) => {
+const syncCommonStateToNativeMobile = (store: Store) => {
   if (NATIVE_MOBILE) {
-    let currentState: RootState
+    let previousState: RootState
 
     store.subscribe(() => {
       const state: RootState = store.getState()
-      const previousState = currentState
-      currentState = state
-      if (
-        !previousState ||
-        clientStoreKeys.some(
-          k =>
-            currentState[k as keyof RootState] !==
-            previousState[k as keyof RootState]
-        )
-      ) {
+
+      // Sync entire commonState initially
+      if (!previousState) {
+        postMessage({
+          type: MessageType.SYNC_COMMON_STATE,
+          state: pick(state, commonStateKeys)
+        })
+
+        previousState = state
+        return
+      }
+
+      // Subsequently only sync the changed slices
+      const stateToSync = pickBy(
+        state,
+        (value, key) =>
+          commonStateKeys.includes(key) &&
+          value !== previousState[key as keyof RootState]
```

Comment:
so if im understanding correctly, only sync the objects at the top level that have changed?

Reply:
Exactly! We could use flatten the state to get even more benefit here